src/tools/interop/idt/discovery/matter_ble.py

```diff
@@ -0,0 +1,108 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import asyncio
+import datetime
+import logging
+import os
+import time
+
+from bleak import AdvertisementData, BleakScanner, BLEDevice
+from bleak.exc import BleakDBusError
+
+
+class MatterBleScanner:
+
+    def __init__(self, artifact_dir: str):
+        self.artifact_dir = artifact_dir
+        self.logger = logging.getLogger(__file__)
+        self.devices_seen_last_time: set[str] = set()
+        self.devices_seen_this_time: set[str] = set()
+        self.throttle_seconds = 1
+        self.error_seconds = 5
+
+    def parse_vid_pid(self, loggable_data: str) -> str:
+        try:
+            vid = loggable_data[8:10] + loggable_data[6:8]
+            pid = loggable_data[12:14] + loggable_data[10:12]
+        except IndexError:
+            self.logger.warning("Error parsing vid / pid from BLE ad data")
+            return ""
+        return f"VID: {vid} PID: {pid}"
+
+    def write_device_log(self, device_name: str, to_write: str) -> None:
+        log_file_name = os.path.join(self.artifact_dir, f"{device_name}.txt")
+        # TODO: Don't call open every time
+        with open(log_file_name, "a+") as log_file:
+            ts = datetime.datetime.now().isoformat(sep=' ', timespec='milliseconds')
+            to_write = f"{ts}\n{to_write}\n\n"
+            log_file.write(to_write)
+            print(to_write)
+
+    @staticmethod
+    def is_matter_device(service_uuid: str) -> bool:
+        is_matter_device = service_uuid.startswith("0000fff6")
+        return is_matter_device
+
+    def handle_device_states(self) -> None:
+        for device_id in self.devices_seen_last_time - self.devices_seen_this_time:
+            to_log = f"LOST {device_id}"
+            self.write_device_log(device_id, to_log)
+        self.devices_seen_last_time = self.devices_seen_this_time
+        self.devices_seen_this_time = set()
+
+    def log_ble_discovery(
```

Comment:
This feels like a higher level analyzer. At the lowest level, we should be capturing raw HCI packets. There should be something processing the stream and identifying the service advertisements.

Reply:
Agreed a better BT capture, likely from the commissioner is needed (e.g. android). As with the DNSSD discovery script, this script is merely meant to be a helpful interactive tool while executing a manual test. More than happy to include a low level script, but I think it would make the most sense in capture?